### ISSUE TITLE
Create a separate component for Split Buttons in Contextual Menu's

### ISSUE THREAD

**Issue by chang47** [microsoft/fluentui#4522]
### Bug Report

__Package version(s)__: 5.80.0

#### Priorities and help requested (not applicable if asking question):

Yes

Requested priority: (Blocking, High, Normal, Low) 

Low 

#### Describe the issue:

The contextual menu is growing too large and there should be some work to componetize and separate the menu items, specifically the contextual menu.

This problem became especially apparent in this PR: 

https://github.com/OfficeDev/office-ui-fabric-react/pull/4353

Where a new limitation introduced by React 16 is preventing us from being able to access the refs of children components by comonentDidUpdate to add event listeners that are not yet supported by React as an example.

This can be solved by abstracting out the Split button menu item from being part of Contextual Menu and be moved to be it's own component with its own state management.

**Comment by Jahnp** [comment]
This sounds reasonable to me, but would love @joschect 's thoughts as codeowner for ContextualMenu. Also flagging this to @dzearing and @micahgodbolt for any thoughts on further extending Button in this way.

**Comment by micahgodbolt** [comment]
blocked by #6134

**Comment by JustSlone** [comment]
This is no longer relevant (or will be fixed) with converged components and our approach on streamlining components going forward
